File: 09-03/example_exp.jl

function myexp(x; tol= 1e-10, maxiter=100)
    term = 1.0
    y = term
    n = 1
    flag = 0
    while flag == 0
        term *= x / n
        y += term
        n += 1
        if abs(term) < tol
            flag = 1
            return y,n-1
        elseif n > maxiter
            flag = -1
            error("Max iterations reached")
        end
    end

end

(exp(100.0) - myexp(100.0, maxiter=2000)[1])/exp(100.0)

File: 09-03/example_exp.py
# make an example pyton implementation of exp(x)

# optional keyword arguments tolerance=1e-10, max_iter=1000
def exp(x, tolerance=1e-10, max_iter=1000):
    y = 1.0
    term = 1.0
    n = 1
    while term > tolerance and n < max_iter:
        term *= x / n
        y += term
        n += 1
    if n == max_iter:
        print("Warning: max_iter reached")
    return y

# test the function
if __name__ == "__main__":  
    import math
    x = 100.0
    print(f"exp({x}) = {exp(x)}")
    print(f"math.exp({x}) = {math.exp(x)}")
    print(f"Difference = {abs(exp(x) - math.exp(x))}")
    #print relative difference
    # print(f"Relative Difference = {abs(exp(x) - math.exp(x)) / math.exp(x)}")